pytorch_lightning/core/lightning.py

```diff
@@ -75,7 +75,7 @@ def __init__(self, *args, **kwargs):
         self._device = torch.device('cpu')
 
         # register all params passed into the child module in __init__
-        self._auto_register_hparams()
+        self._auto_collect_arguments()
```

Comment:
If user now has to call `_auto_register_arguments` method which always calls `_auto_collect_arguments` if needed, why this call is needed there? Maybe some users who want to manually register arguments do not need to collect them.

Reply:
You don't have to, it is just simplification of manual assignment which we would do otherwise manually...

Reply:
@festeh to clarify, this will collect the args but not assign them to the module  directly now. 
it is just internally used to store and restore the model.

Reply:
@awaelchli this is still a side effect,  while previously is was "your hparams will be saved" now it's "all of your args will be saved despite of your consent". As sometimes args could be very heavy, for example if one wants to pass `LightningModule`s for them, this is a valid concern.
 I'll probably have to come with a separate PR which allows users to control what they want to save.